src/snowflake/snowpark/mock/mock_plan.py

```diff
@@ -240,41 +218,30 @@ def calculate_expression(
         )
         return MOCK_FUNCTION_IMPLEMENTATION_MAP[exp.name](output_columns, **kw)
     if isinstance(exp, ListAgg):
-        col = calculate_expression(exp.col, input_data, analyzer)
-        column = input_data[col]
+        column = calculate_expression(exp.col, input_data, analyzer)
         return MOCK_FUNCTION_IMPLEMENTATION_MAP["listagg"](
             [column], is_distinct=exp.is_distinct, delimiter=exp.delimiter
         )
-
-
-def calculate_condition(exp: Expression, dataframe, analyzer, condition=None):
-    if isinstance(exp, Attribute):
-        return analyzer.analyze(exp)
     if isinstance(exp, IsNull):
-        child_condition = calculate_condition(exp.child, dataframe, analyzer, condition)
-        return dataframe[child_condition].isnull()
+        child_condition = calculate_expression(exp.child, input_data, analyzer)
```

Comment:
nit: we shall rename `child_condition` to `child_dataframe` to avoid confusion

Reply:
Changed to `child_column` as the output should be a column